TextExtractor inefficiency solution - performace improved

PrandiniUniPD opened the pull request:
This pull request solve the inefficiency introduced for respecting the OCRInterface ( #24 ) and there is some improvement of the performance.
Also there is the actual code for testing the database but it will be improved.

lucamoroz committed:
Created skeleton for testing the OCR

lucamoroz committed:
updated skeleton

frankplus committed:
few updates to skeleton

added TestInstance class

lucamoroz committed:
updated skeleton, added test class, added Utils class, started testAndReport method and added TODOs

frankplus committed:
Merge remote-tracking branch 'origin/photo_testing' into photo_testing

frankplus committed:
Correct load of test set

frankplus committed:
Rename a method

lucamoroz committed:
updated skeleton, filled some methods

frankplus committed:
Merge remote-tracking branch 'origin/photo_testing' into photo_testing

lucamoroz committed:
fixed get ingredients array from JSON

lucamoroz committed:
new JSON format for a better report

GiovanniFurlan committed:
Better Performance

Deleted InternalStorageClass and all references
Image passed through file path in preferences
text passed through preferences
all SharedPreferences renamed to one "prefs" 
all preferences calls adapted
add tempFileImage() method

GiovanniFurlan committed:
Second Commit

GiovanniFurlan committed:
Add reset preferences on take photo button

-reset all preferences when mButtonTakePhoto is clicked
-preferences default set to null
-more comments

leopardo-rossi-developer committed:
Initial commit

leopardo-rossi-developer committed:
AdapterTestEntry skeleton

Definition of AdapterTestEntry skeleton

lucamoroz committed:
Merge pull request #21 from PrandiniUniPD/photo_testing_JSONReport

new JSON format for a better report

frankplus committed:
Renamed TestInstance2 into TestInstance

stefanoromanello committed:
Fixed ArrayList and ReadFrom file

ArrayList not implemented correctly
Class Utils not used correctly

GiovanniFurlan committed:
Add JSONReportParser

lucamoroz committed:
Fixed getFilePrefix() - getTextFromFile(), updated Utils class, added tests in UtilsTest and PhotoTesterTest

frankplus committed:
Split removing whitespaces using regex

GiovanniFurlan committed:
Comments

lucamoroz committed:
corrected PhotoTesterTest class and UtilsTest class

PietroPrandini committed:
first commit - developing state

PietroPrandini committed:
AdapterTestEntry completed?

PietroPrandini committed:
AdapterTestEntry completed? - clean

PietroPrandini committed:
AdapterTestEntry completed? - clean

PietroPrandini committed:
AdapterTestEntry completed? - clean

PietroPrandini committed:
TestResultActivity - start of the developing

GiovanniFurlan committed:
Merge branch 'photo_texting_presentation_JSONReportParser' into photo_testing_presentation_AdapterTestEntry

leopardo-rossi-developer committed:
Corrections

TODO for the development

leopardo-rossi-developer committed:
Changes

TODO for the development

GiovanniFurlan committed:
Add parseReport() and other changes

- all method to static
-comments
-parseReport()

PietroPrandini committed:
AdapterTestEntry - redefined view

PietroPrandini committed:
AdapterTestEntry - confidence colors corrected

leopardo-rossi-developer committed:
Corrections

Method modifications to improve the code quality

leopardo-rossi-developer committed:
Corrections

Method modification to improve code quality

leopardo-rossi-developer committed:
Corrections

Changes in methods' names

leopardo-rossi-developer committed:
Adapter corrections

Labels' heading removal

PietroPrandini committed:
Merge branch 'photo_texting_presentation_JSONReportParser' into photo_testing_presentation_AdapterTestEntry

PietroPrandini committed:
Merge remote-tracking branch 'origin/CommonDemo' into photo_testing_Better_performance

# Conflicts:
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java
#	app/src/main/java/unipd/se18/ocrcamera/TextExtractor.java

PietroPrandini committed:
Conflicts resolved

PietroPrandini committed:
AsyncTask from code of the group3

PietroPrandini committed:
Restoring from SharedPreferences adjusted

PietroPrandini committed:
Restoring from SharedPreferences (working on)

PietroPrandini committed:
Restoring from SharedPreferences adjusted definitely

PietroPrandini committed:
working on TestResultActivity

PietroPrandini committed:
Merge branch 'photo_testing_presentation' into photo_testing

# Conflicts:
#	.idea/misc.xml
#	.idea/modules.xml
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java
#	app/src/main/java/unipd/se18/ocrcamera/Utils.java
#	app/src/main/res/values/strings.xml
#	app/src/test/java/unipd/se18/ocrcamera/PhotoTesterTest.java
#	app/src/test/java/unipd/se18/ocrcamera/UtilsTest.java

PietroPrandini committed:
Merge remote-tracking branch 'origin/photo_testing_Better_performance' into photo_testing

PietroPrandini committed:
camera activity buttons

lucamoroz committed:
added photo testing from Android/data/unipd.se18.orccamera/files/Pictures

PietroPrandini committed:
Button for launching ocr tests in ResultActivity

PietroPrandini committed:
TestResultActivity xml adjusted

PietroPrandini committed:
AsyncReport in TestResultActivity

PietroPrandini committed:
", " after the words of the ingredients and tags

PietroPrandini committed:
Access granted from a selected directory but "E/PhotoTester: Error decoding JSON"

PietroPrandini committed:
"E/PhotoTester: Error decoding JSON" resolved

PietroPrandini committed:
AsyncTask for tests

PietroPrandini committed:
Landscape xml

PietroPrandini committed:
Logs in ingredientsTextComparison useful to understand what it do

PietroPrandini committed:
ingredientsTextComparison %

frankplus committed:
ingredientsTextComparison change of algorithm

frankplus committed:
Merge remote-tracking branch 'origin/photo_testing' into photo_testing

frankplus committed:
Change confidence data type to float

frankplus committed:
Corrected casting bug

frankplus committed:
Better confidence estimation 

changed ingredientsTextComparison for a better confidence estimation. Now the text is splitted word by word instead of ingredient by ingredient

PietroPrandini committed:
tags in TestResultActivity

lucamoroz committed:
added CountDownLatch to wait for the text extractor to finish

lucamoroz committed:
added multithreading to launch concurrent tests

lucamoroz committed:
recognized test field is now included in testAndReport()

PietroPrandini committed:
Some works to the logs of the PhotoTester and the TextExtractor class

PietroPrandini committed:
Some works to the logs of the PhotoTester and the TextExtractor class

PietroPrandini committed:
Some works to the logs of the PhotoTester and the TextExtractor class

PietroPrandini committed:
Some works to the logs of the PhotoTester and the TextExtractor class

PietroPrandini committed:
Toast of directory for the database data adjusted

frankplus committed:
UI changes: Moved test button in menu

restored floating action button in result activity

PietroPrandini committed:
testAndReport - max concurrent tasks == number of the available cores

lucamoroz reviewed (approved):
The performance is good and the code structure avoid the single point of failure

lucamoroz committed:
added write report to file - after a test the report will be saved to report.txt inside the test directory

frankplus committed:
Change of ingredientsTextComparison algorithm

frankplus committed:
Change of ingredientsTextComparison algorithm

Now confidence estimation is based on the number of characters of each word found

frankplus committed:
Change of ingredientsTextComparison algorithm

For a better estimation

PietroPrandini committed:
Shadow to the TextView of the confidence percentage

frankplus committed:
Change of ingredientsTextComparison algorithm

more optimized and taking into consideration out of order text

stefanoromanello committed:
Added extracted Text to the result of TextActivity

PietroPrandini committed:
test_entry.xml adjusted

frankplus committed:
Change of ingredientsTextComparison algorithm, now using Levenshtein 

implemented Weighted Levenshtein library

PietroPrandini committed:
Same FloatingActionButton in the CameraActivity

PietroPrandini committed:
Merge remote-tracking branch 'origin/photo_testing' into photo_testing

frankplus committed:
Small change

frankplus committed:
removed finish activity

frankplus committed:
bugfix

lucamoroz committed:
Unified TestInstance and TestEntry into TestElement, removed JSONParser class to avoid multiple redundant reading/writing

lucamoroz committed:
added TestElement junit tests, removed useless tests and files

PietroPrandini committed:
Pic preview added to the result activity of the test

PietroPrandini committed:
AdapterTestEntry log and clean

NicoloCervo committed:
added method to get scores for each tag

NicoloCervo committed:
retro compatibility

NicoloCervo committed:
final corrections

frankplus committed:
Added statistics to TestResultActivity

frankplus committed:
Merge branch 'photo_testing_tag_results' into photo_testing

PrandiniUniPD commented:
Some people are working on the app, so the people need of an updated common base to work on it.
Also this pull request is approved by a variety of people.
Coseguently I'm going to merge this pull request into CommonDemo.
